src/tests/src/solver/optim-model-filler/test_componentFiller.cpp

```diff
@@ -218,6 +225,34 @@ BOOST_AUTO_TEST_CASE(var_with_literal_bounds_to_filler__problem_contains_one_var
     BOOST_CHECK_EQUAL(pb->getObjectiveCoefficient(var), 0);
 }
 
+BOOST_AUTO_TEST_CASE(ten_timesteps_var_with_literal_bounds_to_filler__problem_contains_ten_vars)
+{
+    createModelWithOneFloatVar("some_model",
+                               {},
+                               "var1",
+                               literal(-5),
+                               literal(10),
+                               {},
+                               nullptr,
+                               true);
+    createComponent("some_model", "some_component");
+    constexpr unsigned int last_time_step = 9;
+    FillContext ctx{0, last_time_step};
+    buildLinearProblem(ctx);
+    const auto nb_var = ctx.getNumberOfTimestep(); // = 10
+    BOOST_CHECK_EQUAL(pb->variableCount(), nb_var);
+    BOOST_CHECK_EQUAL(pb->constraintCount(), 0);
+    for (unsigned int i = 0; i < nb_var; i++)
+    {
+        auto* var = pb->getVariable("some_component.var1_" + to_string(i));
+        BOOST_CHECK(var);
```

Comment:
```suggestion
        BOOST_REQUIRE(var);
```
will avoid a segfault if `var==nullptr`